pkg/sql/logictest/logic.go

```diff
@@ -1063,10 +1070,23 @@ func (t *logicTest) setUser(user string) func() {
 	addr := t.cluster.Server(t.nodeIdx).ServingSQLAddr()
 	pgURL, cleanupFunc := sqlutils.PGUrl(t.rootT, addr, "TestLogic", url.User(user))
 	pgURL.Path = "test"
-	db, err := gosql.Open("postgres", pgURL.String())
+
+	base, err := pq.NewConnector(pgURL.String())
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	connector := pq.ConnectorWithNoticeHandler(base, func(notice *pq.Error) {
```

Comment:
Is it worth the effort to not set this up if `noticetrace` is false?

Reply:
i think we have to regardless - 'setUser' i think is called before we know whether we're doing a notice trace.

Reply:
sg